files/default/handler/audit_report.rb

```diff
@@ -112,14 +113,15 @@ def load_automate_fetcher
       end
 
       # sets format to json-min when chef-compliance, json when chef-automate
-      def get_opts(format, quiet)
+      def get_opts(format, quiet, attributes)
         output = quiet ? ::File::NULL : $stdout
         Chef::Log.warn "Format is #{format}"
         opts = {
           'report' => true,
           'format' => format,
           'output' => output,
-          'logger' => Chef::Log, # Use chef-client log level for inspec run
+          'logger' => Chef::Log, # Use chef-client log level for inspec run,
+          'attributes' => attributes,
```

Comment:
The attributes key needs to be a symbol, otherwise it's not working; ie. `attributes: attributes,`

As-is, you can see that there's another symbol key that is not assigned the value I'm assigning:

```
From: /opt/chef/embedded/lib/ruby/gems/2.4.0/gems/inspec-1.30.0/lib/inspec/runner.rb @ line 124 Inspec::Runner#load_attributes:

    120: def load_attributes(options)
    121:   options[:attributes] ||= {}
    122:   require 'pry';binding.pry
    123:
 => 124:   secrets_targets = options[:attrs]
    125:   return options[:attributes] if secrets_targets.nil?
    126:
    127:   secrets_targets.each do |target|
    128:     secrets = Inspec::SecretsBackend.resolve(target)
    129:     if secrets.nil?
    130:       raise Inspec::Exceptions::SecretsBackendNotFound,
    131:             "Unable to find a parser for attributes file #{target}. " \
    132:             'Check to make sure the file exists and has the appropriate extension.'
    133:     end
    134:
    135:     next if secrets.attributes.nil?
    136:     options[:attributes].merge!(secrets.attributes)
    137:   end
    138:
    139:   options[:attributes]
    140: end

[1] pry(#<Inspec::Runner>)> options
=> {"report"=>true,
 "format"=>"json",
 "output"=>"/dev/null",
 "logger"=>Chef::Log,
 "attributes"=>{"password"=>"bob"},
 :logger=>#<Logger:0x00000004f37010 @default_formatter=#<Logger::Formatter:0x00000004f36fc0 @datetime_format=nil>, @formatter=nil, @level=0, @logdev=nil, @progname=nil>,
 :attributes=>{}}
[2] pry(#<Inspec::Runner>)>
```

When line 124 is changed to `attributes: attributes,`:

```
From: /opt/chef/embedded/lib/ruby/gems/2.4.0/gems/inspec-1.30.0/lib/inspec/runner.rb @ line 124 Inspec::Runner#load_attributes:

    120: def load_attributes(options)
    121:   options[:attributes] ||= {}
    122:   require 'pry';binding.pry
    123:
 => 124:   secrets_targets = options[:attrs]
    125:   return options[:attributes] if secrets_targets.nil?
    126:
    127:   secrets_targets.each do |target|
    128:     secrets = Inspec::SecretsBackend.resolve(target)
    129:     if secrets.nil?
    130:       raise Inspec::Exceptions::SecretsBackendNotFound,
    131:             "Unable to find a parser for attributes file #{target}. " \
    132:             'Check to make sure the file exists and has the appropriate extension.'
    133:     end
    134:
    135:     next if secrets.attributes.nil?
    136:     options[:attributes].merge!(secrets.attributes)
    137:   end
    138:
    139:   options[:attributes]
    140: end

[1] pry(#<Inspec::Runner>)> options
=> {"report"=>true,
 "format"=>"json",
 "output"=>"/dev/null",
 "logger"=>Chef::Log,
 :attributes=>{"password"=>"bob"},
 :logger=>#<Logger:0x00000003d4acd0 @default_formatter=#<Logger::Formatter:0x00000003d4ac80 @datetime_format=nil>, @formatter=nil, @level=0, @logdev=nil, @progname=nil>}
[2] pry(#<Inspec::Runner>)>
```

With the change, my inspec control test works as expected, without, I get nil for the attribute value.

As I mentioned, this is because of the key being symbol: https://github.com/chef/inspec/blob/master/lib/inspec/runner.rb#L121